mycli/output_formatter/output_formatter.py

```diff
@@ -0,0 +1,97 @@
+    supported_formats as terminaltables_formats)
+
+OutputFormatHandler = namedtuple(
+    'OutputFormatHandler',
+    'format_name preprocessors formatter formatter_args')
+
+
+class OutputFormatter(object):
+    """A class with a standard interface for various formatting libraries."""
+
+    _output_formats = {}
+
+    def __init__(self, format_name=None):
+        """Register the supported output formats."""
+        self._format_name = format_name
+
+    def set_format_name(self, format_name):
+        """Set the OutputFormatter's default format."""
+        if format_name in self.supported_formats():
+            self._format_name = format_name
+        else:
+            raise ValueError('unrecognized format_name: {}'.format(
+                format_name))
+
+    def get_format_name(self):
+        """Get the OutputFormatter's default format."""
+        return self._format_name
+
+    def supported_formats(self):
+        """Return the supported output format names."""
+        return tuple(self._output_formats.keys())
+
+    @classmethod
+    def register_new_formatter(cls, format_name, handler, preprocessors=None,
+                               kwargs=None):
+        """Register a new formatter to format the output."""
+        cls._output_formats[format_name] = OutputFormatHandler(
+            format_name, preprocessors, handler, kwargs)
+
+    def format_output(self, data, headers, format_name=None, **kwargs):
+        """Format the headers and data using a specific formatter.
+
+        *format_name* must be a formatter available in `supported_formats()`.
+
+        All keyword arguments are passed to the specified formatter.
+        """
+        format_name = format_name or self._format_name
+        if format_name not in self.supported_formats():
+            raise ValueError('unrecognized format: {}'.format(format_name))
+
+        (_, preprocessors, formatter, fkwargs) = self._output_formats[format_name]
+        fkwargs.update(kwargs)
+        if preprocessors:
+            for f in preprocessors:
+                data, headers = f(data, headers, **fkwargs)
+        return formatter(data, headers, **fkwargs)
+
+
+OutputFormatter.register_new_formatter('csv', delimiter_adapter,
+                                       delimiter_preprocessors,
+                                       {'missing_value': '<null>'})
+OutputFormatter.register_new_formatter('tsv', delimiter_adapter,
+                                       delimiter_preprocessors,
+                                       {'missing_value': '<null>',
+                                        'delimiter': '\t'})
+OutputFormatter.register_new_formatter('expanded', expanded_table,
+                                       (override_missing_value,
+                                        convert_to_string),
+                                       {'missing_value': '<null>',
+                                        'delimiter': '\t'})
```

Comment:
Yup. That is indeed a copy/paste error. 😞 

I should have moved expanded.py to the output_formatter package. That's a good call.